deployment/README-TLS-Enablement.md

```diff
@@ -0,0 +1,45 @@
+
+Flavia, I wanted to list the changes I needed to do to enable TLS and 'cross cluster' access via a route.  
+Please contact me if you need assistance incorporating this into your overall deployment process.
+
+## Update the caikit configuration
+- The .../demo/server/config.yml file must be updated to include where to find the server certificate and key. Note, the locations under /mnt are refenced in the yaml described below.  The image must then be re-built after updating the config.yml.
+
+       ```
+       # Caikit model configuration
+       # Relative paths are relative to where the config is loaded
+       runtime:
+           library: caikit_embeddings
+           grpc: {"enabled": True}
+           http: {"enabled": True}
+           # All models in this directory are loaded at boot time
+           local_models_dir: "../models"
+           tls:
+               server:
+                   key: /mnt/tls.key
+                   cert: /mnt/tls.crt
+       ```
+## Deploying the certificates
+
+- The TLS certificates and keys are deployed within a OCP secret.  
+- Deploy the .../deployment/embeddings-caikit-secret.yaml file.
+- I had to manually create the certificates and keys, insuring the correct Subject Alternative Names (SANs) were added to the certificates. The SANs are set up to work with our routes and DNS.  
+- If you wanted to create your own certificates/keys you can do the following:
+```
+    1. Create a server-ext.cnf file with the following contents (these are for our clusters). 
+	    subjectAltName=DNS:*.apps.fmaas-devstage-backend.fmaas.res.ibm.com,DNS:*.fmaas-devstage-backend.fmaas.res.ibm.com,DNS:*.fmaas-backend.fmaas.res.ibm.com,DNS:*.apps.fmaas-backend.fmaas.res.ibm.com,DNS:*.cash.sl.cloud9.ibm.com,DNS:*.apps.c1-ocp.sl.cloud9.ibm.com,IP:0.0.0.0.
+    2. Generate CA's private key and self-signed certificate
+	    openssl req -x509 -newkey rsa:4096 -days 3650 -nodes -keyout tls/ca-key.pem -out tls/ca-cert.pem -subj "/C=US/ST=New York/L=Yorktown Heights/O=IBM/OU=IBM Research/CN=embeddings-ca"
+    3. Generate web server's private key and certificate signing request (CSR)
+	    openssl req -newkey rsa:4096 -nodes -keyout tls/server-key.pem -out tls/server-req.pem -subj "/C=US/ST=New York/L=Yorktown Heights/O=IBM/OU=IBM Research/CN=*.cash.sl.cloud9.ibm.com/emailAddress=cash@us.ibm.com"
+    4. Use CA's private key to sign web server's CSR and get back the signed certificate
+	    openssl x509 -req -in tls/server-req.pem -days 730 -CA tls/ca-cert.pem -CAkey tls/ca-key.pem -CAcreateserial -out tls/server-cert.pem -extfile ./server-ext.cnf
+    5. Then copy the ca-cert, server-cert, and server key to secret
+```
+
+## Apply deployment,  service, and gRPC route
+- Apply .../deployment/deployment-caikit-embeddings.yaml is a 'deploymentConfig' I converted to a 'deployment'
+  - Note: The TLS secrets MUST BE mounted within this file!!!
+- Apply .../service-caikit-embeddings.yaml (previously existed)
```

Comment:
same here.  "previously existed" is like giving directions "turn left where there used to be stop sign" to somebody that never knew the stop sign.

So again:  good notes, but probably better to describe the current and not how we got here